django/db/models/sql/query.py

```diff
@@ -1768,9 +1766,6 @@ def add_immediate_loading(self, field_names):
         """
         existing, defer = self.deferred_loading
         field_names = set(field_names)
-        if 'pk' in field_names:
-            field_names.remove('pk')
-            field_names.add(self.get_meta().pk.name)
```

Comment:
I'm fairly certain you can't defer primary keys. That being said calling `.only('pk', 'id')` does work because the data is tracked as a dictionary with the value as a set. I added a couple more examples to the tests.